generative/maisi/scripts/find_masks.py

```diff
@@ -0,0 +1,122 @@
+
+    for _k in range(len(body_region)):
+        region = body_region[_k].lower()
+
+        idx = None
+        if "head" in region:
+            idx = 0
+        elif "chest" in region or "thorax" in region or "chest/thorax" in region:
+            idx = 1
+        elif "abdomen" in region:
+            idx = 2
+        elif "pelvis" in region or "lower" in region or "pelvis/lower" in region:
+            idx = 3
+        else:
+            raise ValueError("Input region information is incorrect.")
+
+        body_region_indices.append(idx)
+
+    return body_region_indices
+
+
+def find_masks(
+    body_region: str | list[str],
+    anatomy_list: int | list[int],
+    spacing: list[float],
+    output_size: list[int],
+    check_spacing_and_output_size: bool = False,
+    database_filepath: str = "./database.json",
+    mask_foldername: str = "./masks",
+):
+    if type(body_region) is str:
+        body_region = [body_region]
+
+    body_region = convert_body_region(body_region)
+
+    if type(anatomy_list) == int:
+        anatomy_list = [anatomy_list]
+
+    if not os.path.isfile(database_filepath):
+        raise ValueError(f"Please download {database_filepath}.")
+
+    if not os.path.exists(mask_foldername):
+        zip_file_path = mask_foldername + ".zip"
+
+        if not os.path.isfile(zip_file_path):
+            raise ValueError(f"Please downloaded {zip_file_path}.")
+
+        with zipfile.ZipFile(zip_file_path, "r") as zip_ref:
+            print(mask_foldername)
+            zip_ref.extractall(path="./datasets")
+        print(f"Unzipped {zip_file_path} to {mask_foldername}.")
+
+    with open(database_filepath, "r") as f:
+        db = json.load(f)
+
+    candidates = []
+    for _i in range(len(db)):
+        _item = db[_i]
+        if not set(anatomy_list).issubset(_item["label_list"]):
+            continue
+
+        top_index = [index for index, element in enumerate(_item["top_region_index"]) if element != 0]
+        top_index = top_index[0]
+        bottom_index = [index for index, element in enumerate(_item["bottom_region_index"]) if element != 0]
+        bottom_index = bottom_index[0]
+
+        flag = False
+        for _idx in body_region:
+            if _idx > bottom_index or _idx < top_index:
+                flag = True
+
+        # check if candiate mask contains tumors
+        for tumor_label in [23, 24, 26, 27, 128]:
+            # we skip those mask with tumors if users do not provide tumor label in anatomy_list
+            if tumor_label not in anatomy_list and tumor_label in _item["label_list"]:
+                flag = True
+
+        if check_spacing_and_output_size:
+            # check if the output_size and spacing are same as user's input
+            for axis in range(3):
+                if _item["dim"][axis] != output_size[axis] or _item["spacing"][axis] != spacing[axis]:
+                    flag = True
+
+        if flag == True:
+            continue
+
+        candidate = {}
```

Comment:
The following may improve code readability:

```python
candidate = {
  "pseudo_label": ...
 }
if "label_filename" in _item:
  candidate["label"] = os.path.join(mask_foldername, _item["label_filename"])
```

Also, for values that are file paths, I would suggest to use key names such as `**_path` or `**_filepath` to indicate these are actually no the contents.

Reply:
thanks, will do